c++/src/kj/string.h

```diff
@@ -378,27 +358,18 @@ class ConstString {
   inline bool operator!=(decltype(nullptr)) const { return content.size() > 1; }
 
   inline bool operator==(const StringPtr& other) const { return StringPtr(*this) == other; }
-#if !__cpp_impl_three_way_comparison
-  inline bool operator!=(const StringPtr& other) const { return StringPtr(*this) != other; }
-#endif
   inline bool operator< (const StringPtr& other) const { return StringPtr(*this) <  other; }
   inline bool operator> (const StringPtr& other) const { return StringPtr(*this) >  other; }
   inline bool operator<=(const StringPtr& other) const { return StringPtr(*this) <= other; }
   inline bool operator>=(const StringPtr& other) const { return StringPtr(*this) >= other; }
 
   inline bool operator==(const String& other) const { return StringPtr(*this) == StringPtr(other); }
-#if !__cpp_impl_three_way_comparison
-  inline bool operator!=(const String& other) const { return StringPtr(*this) != StringPtr(other); }
-#endif
   inline bool operator< (const String& other) const { return StringPtr(*this) <  StringPtr(other); }
   inline bool operator> (const String& other) const { return StringPtr(*this) >  StringPtr(other); }
   inline bool operator<=(const String& other) const { return StringPtr(*this) <= StringPtr(other); }
   inline bool operator>=(const String& other) const { return StringPtr(*this) >= StringPtr(other); }
 
   inline bool operator==(const ConstString& other) const { return StringPtr(*this) == StringPtr(other); }
-#if !__cpp_impl_three_way_comparison
-  inline bool operator!=(const ConstString& other) const { return StringPtr(*this) != StringPtr(other); }
-#endif
   inline bool operator< (const ConstString& other) const { return StringPtr(*this) <  StringPtr(other); }
```

Comment:
should we have `<=>` operator for everything instead of these then?

Reply:
I definitely think so -- it'll reduce boilerplate across all of our codebases. I suspect there may be debate over it, so held off for now.